tests/test_mutations.py

```diff
@@ -470,3 +470,57 @@ def test_keep_mutation_parent_zero_rate(self):
         t1.provenances.clear()
         t2.provenances.clear()
         self.assertEqual(t1, t2)
+
+
+def mutate_map(tables, mutation_map, seed):
+    """
+    Parallel implementation of the C code which should produce precisely the same
+    results.
+    """
+    rng = msprime.RandomGenerator(seed)
+    time = tables.nodes.time
+    for edge in tables.edges:
+        branch_length = time[edge.parent] - time[edge.child]
+        index = np.searchsorted(mutation_map.position, edge.left)
```

Comment:
This is my suggested implementation @petrelharp, does it look OK to you? Basically we can replace ``searchsorted`` with bsearch and it'll be basically identical after that.